### PR TITLE
Problem auth

### PR DESCRIPTION
**Pull request by Changron**
1. Add loading info when opening contest info modal or register modal
2. Add auth check of problmes when creating or editing contest 
   admin: see all
   owner: see self create or visible problem
   coowner: same as contest owner

### CODE REVIEW

**Comment by nickelcheng** [comment]
I got an error
screenshot from 2015-04-21 01 38 45

**Comment by Changron** [comment]
fixed

File: contest/forms.py [hunk 1]
```diff
@@ -22,12 +22,35 @@
...
+from django.db.models import Q
 
 class ContestForm(forms.ModelForm):
     def __init__(self, *args, **kwargs):
         super(ContestForm, self).__init__(*args, **kwargs)
         # access object through self.instance...
+        initial = kwargs.get('initial',{})
+        user = initial.get('user',User())
+        method = initial.get('method','')
         self.fields['coowner'].queryset = User.objects.exclude(user_level=User.USER)
```
Comment (nickelcheng) [inline contest/forms.py]: Please exclude owner as well.

File: contest/forms.py [hunk 2]
```diff
@@ -22,12 +22,35 @@
...
+            contest_id = initial.get('id',0)
+            # if user not is admin
+            # get all problem when user is admin
+            if not user.has_admin_auth():
+                # edit contest
+                if contest_id:
+                    contest = Contest.objects.get(pk = contest_id)
+                    contest_problems = contest.problem.all().distinct()
+                    self.fields['problem'].queryset = Problem.objects.filter(
+                            Q(visible = True)|Q(owner = user)).distinct() | contest_problems
```
Comment (nickelcheng) [inline contest/forms.py]: This seems to be a good solution.

**Comment by nickelcheng** [comment]
+1

**Comment by Changron** [comment]
fixed

**Comment by bruce3557** [comment]
+1, going to merge

**Pull request merged**